templates/embeds/google-drive-file.php

```diff
@@ -9,10 +9,18 @@
 	// prevent direct access to this file.
 	exit;
 }
-?>
-<div style="border: 1px solid #000; text-align: center;">
-	<a href="<?php echo esc_url( $drive_file_url ); ?>" title="<?php esc_attr_e( 'Open the Shared Document', 'rt-google-embeds' ); ?>" target="_blank" rel="noopener noreferrer" style="color: #cd2653;">
-		<?php esc_html_e( 'Open Shared Document.', 'rt-google-embeds' ); ?>
-	</a>
-	<img src="<?php echo esc_url( $thumbnail_url ); ?>" alt="<?php esc_attr_e( 'Shared Document Preview', 'rt-google-embeds' ); ?>" style="border: 1px solid #eee; margin: 15px auto; display: block;" />
-</div>
+if ( file_exists( $cached_file ) ) : ?>
```

Comment:
There won't be a `cached_file` URL here, as if the network request fails, `get_thumbnail_url` function itself will return cached file URL.

Reply:
Instead of adding https://github.com/rtCamp/wp-google-drive/pull/6/files#diff-59c6e676df5dade9eec1de91f5c5b44fR252-R253 lines, you'll only use the URL retrieved from `get_thumbnail_url` function, that function will decide whether to use cached image or network image URL.

Whenever network request is successful, you update the cached image. And when network request fails, you return the cached image URL.